File: email-validation-3-nodejs/SOAP/ev3_response.js
export class ErrorModel {
    constructor(data = {}) {
        this.Type = data.Type || null;
        this.TypeCode = data.TypeCode || null;
        this.Desc = data.Desc || null;
        this.DescCode = data.DescCode || null;
        this.Number = data.Number || null;
    }

    toString() {
        return `Error: Type = ${this.Type}, TypeCode = ${this.TypeCode}, Desc = ${this.Desc}, DescCode = ${this.DescCode}`;
    }
}

export class ValidateEmailInfo {
    constructor(data = {}) {
        this.Score = data.Score || null;
        this.IsDeliverable = data.IsDeliverable || null;
        this.EmailAddressIn = data.EmailAddressIn || null;
        this.EmailAddressOut = data.EmailAddressOut || null;
        this.EmailCorrected = data.EmailCorrected || null;
        this.Box = data.Box || null;
        this.Domain = data.Domain || null;
        this.TopLevelDomain = data.TopLevelDomain || null;
        this.TopLevelDomainDescription = data.TopLevelDomainDescription || null;
        this.IsSMTPServerGood = data.IsSMTPServerGood || null;
        this.IsCatchAllDomain = data.IsCatchAllDomain || null;
        this.IsSMTPMailBoxGood = data.IsSMTPMailBoxGood || null;
        this.WarningCodes = data.WarningCodes || null;
        this.WarningDescriptions = data.WarningDescriptions || null;
        this.NotesCodes = data.NotesCodes || null;
        this.NotesDescriptions = data.NotesDescriptions || null;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        const error = this.Error ? this.Error.toString() : "None";
        return `ValidateEmailInfo: Score = ${this.Score}, IsDeliverable = ${this.IsDeliverable}, ` +
            `EmailAddressIn = ${this.EmailAddressIn}, EmailAddressOut = ${this.EmailAddressOut}, ` +
            `EmailCorrected = ${this.EmailCorrected}, Box = ${this.Box}, Domain = ${this.Domain}, ` +
            `TopLevelDomain = ${this.TopLevelDomain}, TopLevelDomainDescription = ${this.TopLevelDomainDescription}, ` +
            `IsSMTPServerGood = ${this.IsSMTPServerGood}, IsCatchAllDomain = ${this.IsCatchAllDomain}, ` +
            `IsSMTPMailBoxGood = ${this.IsSMTPMailBoxGood}, WarningCodes = ${this.WarningCodes}, ` +
            `WarningDescriptions = ${this.WarningDescriptions}, NotesCodes = ${this.NotesCodes}, ` +
            `NotesDescriptions = ${this.NotesDescriptions}, Error = ${error}`;
    }
}

export class EV3Response {
    constructor(data = {}) {
        this.LicenseKey = data.LicenseKey || null;
        this.EmailAddress = data.EmailAddress || null;
        this.AllowCorrections = data.AllowCorrections || null;
        this.ValidateEmailInfo = data.ValidateEmailInfo ? new ValidateEmailInfo(data.ValidateEmailInfo) : null;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        const validateEmailInfo = this.ValidateEmailInfo ? this.ValidateEmailInfo.toString() : "None";
        const error = this.Error ? this.Error.toString() : "None";
        return `EV3Response: LicenseKey = ${this.LicenseKey}, EmailAddress = ${this.EmailAddress}, ` +
            `AllowCorrections = ${this.AllowCorrections}, ValidateEmailInfo = ${validateEmailInfo}, Error = ${error}`;
    }
}

export default EV3Response;

File: email-validation-3-dot-net/REST/EV3Response.cs
using System.Runtime.Serialization;

namespace email_validation_3_dot_net.REST
{
    /// <summary>
    /// Represents the combined request and response data for performing comprehensive
    /// email address validation using the Email Validation 3 REST API.
    /// </summary>
    [DataContract]
    public class EV3Response
    {
        public string LicenseKey { get; set; }
        public string EmailAddress { get; set; }
        public string AllowCorrections { get; set; }
        public ValidateEmailInfo ValidateEmailInfo { get; set; }
        public Error Error { get; set; }
    }

    /// <summary>
    /// Represents the detailed results of any email validation operation in the Email Validation 3 REST API.
    /// </summary>
    [DataContract]
    public class ValidateEmailInfo
    {
        public int Score { get; set; }
        public string IsDeliverable { get; set; }
        public string EmailAddressIn { get; set; }
        public string EmailAddressOut { get; set; }
        public bool EmailCorrected { get; set; }
        public string Box { get; set; }
        public string Domain { get; set; }
        public string TopLevelDomain { get; set; }
        public string TopLevelDomainDescription { get; set; }
        public string IsSMTPServerGood { get; set; }
        public string IsCatchAllDomain { get; set; }
        public string IsSMTPMailBoxGood { get; set; }
        public string WarningCodes { get; set; }
        public string WarningDescriptions { get; set; }
        public string NotesCodes { get; set; }
        public string NotesDescriptions { get; set; }
        public Error Error { get; set; }
        public override string ToString()
        {
            string error = Error != null ? Error.ToString() : "None";
            return $"ValidateEmailInfo: Score = {Score}, IsDeliverable = {IsDeliverable}, " +
                   $"EmailAddressIn = {EmailAddressIn}, EmailAddressOut = {EmailAddressOut}, " +
                   $"EmailCorrected = {EmailCorrected}, Box = {Box}, Domain = {Domain}, " +
                   $"TopLevelDomain = {TopLevelDomain}, TopLevelDomainDescription = {TopLevelDomainDescription}, " +
                   $"IsSMTPServerGood = {IsSMTPServerGood}, IsCatchAllDomain = {IsCatchAllDomain}, " +
                   $"IsSMTPMailBoxGood = {IsSMTPMailBoxGood}, WarningCodes = {WarningCodes}, " +
                   $"WarningDescriptions = {WarningDescriptions}, NotesCodes = {NotesCodes}, " +
                   $"NotesDescriptions = {NotesDescriptions}, Error = {error}";
        }
    }
    /// <summary>
    /// Represents error information in the Lead Validation International REST API response.
    /// </summary>
    [DataContract]
    public class Error
    {
        public string Type { get; set; }
        public string TypeCode { get; set; }
        public string Desc { get; set; }
        public string DescCode { get; set; }

        public override string ToString()
        {
            return $"Error: Type = {Type}, TypeCode = {TypeCode}, Desc = {Desc}, DescCode = {DescCode}";
        }
    }
}